### ISSUE TITLE
macro calls allowed @example blocks?

### ISSUE THREAD

**Issue by ablaom** [JuliaDocs/Documenter.jl#1128]
I have struggled a couple of time when I have a macro call in an `@example` block. I'm guessing the issue is that code from this block is wrapped inside a baremodule, which has no `eval` method. Sometimes adding `import Base.eval` in a `@setup` block helps, but here is one case where this does not seem to work:

**The failed travis build:**
https://travis-ci.com/alan-turing-institute/MLJ.jl/jobs/234035991

**The offending code.**

```@setup 7
import Base.eval
using MLJ
MLJ.color_off()
```
```@example 7
using MLJ
X = (age    = [23, 45, 34, 25, 67],
     gender = categorical(['m', 'm', 'f', 'm', 'f']));
height = [67.0, 81.5, 55.6, 90.0, 61.1]; nothing # hide
```
```@example 7
pipe = @pipeline MyPipe(X -> coerce(X, :age=>Continuous),
                        hot = OneHotEncoder(),
                        knn = KNNRegressor(K=3),
                        target = UnivariateStandardizer())
params(pipe)
```

**Comment by mortenpi** [comment]
Sorry for a very late reply here. But I don't think it's the macros causing the issue here. In this case, it just seems that you don't have `OneHotEncoder` in the local scope, so [the `pipe = @pipeline MyPipe...` line fails to run](https://travis-ci.com/alan-turing-institute/MLJ.jl/jobs/234035991#L417-L428)?

**Comment by ablaom** [comment]
No, I don't think that is the problem. The code above runs fine in the REPL or wrapped in a module:

```julia
julia> module Junk
       
       using MLJ
       X = (age    = [23, 45, 34, 25, 67],
       gender = categorical(['m', 'm', 'f', 'm', 'f']));
       height = [67.0, 81.5, 55.6, 90.0, 61.1]; nothing # hide
       pipe = @pipeline MyPipe(X -> coerce(X, :age=>Continuous),
       hot = OneHotEncoder(),
       knn = KNNRegressor(K=3),
       target = UnivariateStandardizer())
       
       @show params(pipe)
       
       end

params(pipe) = (hot = (features = Symbol[], drop_last = false, ordered_factor = true), knn = (K = 3, metric = MLJModels.KNN.euclidean, kernel = MLJModels.KNN.reciprocal), target = NamedTuple())
Main.Junk
```

```julia
julia> Pkg.installed()
Dict{String,Union{Nothing, VersionNumber}} with 1 entry:
  "MLJ" => v"0.4.0"

julia> versioninfo()
Julia Version 1.1.1
Commit 55e36cc (2019-05-16 04:10 UTC)
Platform Info:
  OS: macOS (x86_64-apple-darwin15.6.0)
  CPU: Intel(R) Core(TM) i7-8850H CPU @ 2.60GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-6.0.1 (ORCJIT, skylake)
Environment:
  JULIA_PATH = /Applications/Julia-1.1.app/Contents/Resources/julia/bin/julia
```

**Comment by odow** [comment]
Is this still a problem @ablaom? The MWE doesn't work anymore because KNNRegressor isn't defined.

FWIW, JuMP uses macros extensively in `@example` blocks, so I'm not sure what the problem is.

**Comment by ablaom** [comment]
Okay, some time has passed since I looked at this. 

We have since learned that the `@pipeline` macro lacks certain robustness with respect to scope more generally, because of `eval`s in the macro definition. A pending PR replaces it with a non-macro version. 

I think it's a good bet that the issue is with `@pipeline` and not with  Documenter and am going to close this.

@mortenpi @odow Thanks for looking into this.
